signatures/version1.go

```diff
@@ -80,58 +92,40 @@ func Verify(signature string, signatureData *SignatureData) (*x509.Certificate,
 		return nil, err
 	}
 
-	certBytes := []byte(signatureData.Certificate)
+	cfInstanceCertContentsBytes := []byte(signatureData.CFInstanceCertContents)
 	var block *pem.Block
 	var result error
 	for {
-		block, certBytes = pem.Decode(certBytes)
+		block, cfInstanceCertContentsBytes = pem.Decode(cfInstanceCertContentsBytes)
 		if block == nil {
 			break
 		}
-		clientCerts, err := x509.ParseCertificates(block.Bytes)
+		instanceCerts, err := x509.ParseCertificates(block.Bytes)
 		if err != nil {
 			result = multierror.Append(result, err)
 			continue
 		}
-		for _, clientCert := range clientCerts {
-			publicKey, ok := clientCert.PublicKey.(*rsa.PublicKey)
+		for _, instanceCert := range instanceCerts {
+			publicKey, ok := instanceCert.PublicKey.(*rsa.PublicKey)
 			if !ok {
-				result = multierror.Append(result, fmt.Errorf("not an rsa public key, it's a %t", clientCert.PublicKey))
+				result = multierror.Append(result, fmt.Errorf("not an rsa public key, it's a %t", instanceCert.PublicKey))
 				continue
 			}
-
-			if err := rsa.VerifyPSS(publicKey, crypto.SHA256, signatureData.hash(), signatureBytes, nil); err != nil {
+			hash, err := signatureData.hash()
+			if err != nil {
+				multierror.Append(err)
+				continue
+			}
+			if err := rsa.VerifyPSS(publicKey, crypto.SHA256, hash, signatureBytes, nil); err != nil {
 				result = multierror.Append(result, err)
 				continue
 			}
 			// Success
-			return clientCert, nil
+			return instanceCert, nil
 		}
 	}
 	if result == nil {
-		return nil, fmt.Errorf("no matching client certificate found for %s in %s", signature, signatureData.Certificate)
+		return nil, fmt.Errorf("no matching certificate found for %s in %s", signature, signatureData.CFInstanceCertContents)
 	}
 	return nil, result
 }
-
-func IsIssuer(pathToCACert string, clientCert *x509.Certificate) (bool, error) {
```

Comment:
This was moved to `util` because it actually doesn't relate to `signatures` at all so didn't really fit here well.